docs/content/docs/connectors/table/firehose.md

```diff
@@ -0,0 +1,312 @@
+under the License.
+-->
+
+# Amazon Kinesis Data Firehose SQL Connector
+
+{{< label "Sink: Streaming Append Mode" >}}
+
+The Firehose connector allows for writing data into [Amazon Kinesis Data Firehose (KDF)](https://aws.amazon.com/kinesis/data-firehose/).
+
+Dependencies
+------------
+
+{{< sql_download_table "aws-kinesis-firehose" >}}
+
+How to create a Kinesis data firehose table
+-----------------------------------------
+
+Follow the instructions from the [Amazon KDF Developer Guide](https://docs.aws.amazon.com/ses/latest/dg/event-publishing-kinesis-analytics-firehose-stream.html) to set up a firehose stream.
+The following example shows how to create a table backed by a Kinesis data firehose delivery stream with minimum required options:
+
+```sql
+CREATE TABLE FirehoseTable (
+  `user_id` BIGINT,
+  `item_id` BIGINT,
+  `category_id` BIGINT,
+  `behavior` STRING
+)
+WITH (
+  'connector' = 'firehose',
+  'delivery-stream' = 'user_behavior',
+  'aws.region' = 'us-east-2',
+  'format' = 'csv'
+);
+```
+
+Connector Options
+-----------------
+
+<table class="table table-bordered">
+    <thead>
+    <tr>
+      <th class="text-left" style="width: 25%">Option</th>
+      <th class="text-center" style="width: 8%">Required</th>
+      <th class="text-center" style="width: 7%">Default</th>
+      <th class="text-center" style="width: 10%">Type</th>
+      <th class="text-center" style="width: 50%">Description</th>
+    </tr>
+    <tr>
+      <th colspan="5" class="text-left" style="width: 100%">Common Options</th>
+    </tr>
+    </thead>
+    <tbody>
+    <tr>
+      <td><h5>connector</h5></td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Specify what connector to use. For Kinesis Firehose use <code>'firehose'</code>.</td>
+    </tr>
+    <tr>
+      <td><h5>delivery-stream</h5></td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>Name of the Kinesis data firehose delivery stream backing this table.</td>
+    </tr>
+    <tr>
+      <td><h5>format</h5></td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>The format used to deserialize and serialize Kinesis data firehose records. See <a href="#data-type-mapping">Data Type Mapping</a> for details.</td>
+    </tr>
+    <tr>
+      <td><h5>aws.region</h5></td>
+      <td>required</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>The AWS region where the delivery stream is defined. This option is required for <code>KinesisDataFirehoseSink</code> creation.</td>
+    </tr>
+    <tr>
+      <td><h5>aws.endpoint</h5></td>
+      <td>optional</td>
+      <td style="word-wrap: break-word;">(none)</td>
+      <td>String</td>
+      <td>The AWS endpoint for Kinesis Firehose.</td>
+    </tr>
+    <tr>
+      <td><h5>aws.trust.all.certificates</h5></td>
```

Comment:
This line is inconsistent with others, is it correct? There is a mixture of `trust-all-certificates` and `trustAllCertificates` in the options

Reply:
I am sorry, I can't get why it is inconsistent. I couldn't find the `trust-all-certificates` or the `trustAllCertificates` in this document or in the firehose/aws options in the implementation.

Reply:
I did not mean this one particularly. I was pointing out we have mixed naming conventions for the keys:
- `aws.credentials.role.sessionName`
- `aws.credentials.basic.accesskeyid`
- `sink.http-client.read-timeout`
-  `aws.trust.all.certificates`

Why so many? Are the docs correct?

Reply:
I agree!
Unfortunately the docs are correct!
The options were picked to match those of datastream configuration so they are mapped directly to options in `AWSConfigConstants` class (i.e. `aws.credentials.role.sessionName` `aws.credentials.basic.accesskeyid` `aws.http-client.read-timeout` `aws.trust.all.certificates`).
we can follow up with another PR to: 
- enforce consistency on table options to a specific convention and map them
- or enforce consistency on both table options and datastreams by changing Config class. 
However I am not sure what convention to follow since there is not a specific one in the [legacy connector](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/connectors/table/kinesis/#connector-options)

Also since the authentication options are now shared between all aws table connectors, this would impose additional work to isolate the options/add fallbacks from legacy connector to keep it backward compatible. What are your thoughts about it?

Reply:
Whatever we do, we have to retain backwards compatibility. It is too late to make changes now without getting an exception since this is a public API change. This does not need to block the docs, since the docs are correct according to the implementation. 

Which fields have we added for 1.15 that are not already used by the legacy connector?

Reply:
We added in 1.15 
- all the async sink config options (i.e `sink.*`) except for `sink.partitioner-field-delimiter` and `sink.partitioner` for KDS as they were already there.
- only the `aws.trust.all.certificates` auth option as `sink.producer.verify-certificate` was used in the legacy connector.

the new sink options all follow a hyphen convention following the existing `sink.partitioner-field-delimiter` option.